staging/src/k8s.io/kubectl/pkg/cmd/autoscale/autoscale.go

```diff
@@ -220,10 +231,18 @@ func (o *AutoscaleOptions) Run() error {
 
 		// handles the creation of HorizontalPodAutoscaler objects for both v2 and v1 APIs.
 		// If v2 API fails, try to create and handle HorizontalPodAutoscaler using v1 API
-		hpaV2 := o.createHorizontalPodAutoscalerV2(info.Name, mapping)
+		hpaV2, err := o.createHorizontalPodAutoscalerV2(info.Name, mapping)
+		if err != nil {
+			return fmt.Errorf("failed to create v2 HorizontalPodAutoscaler: %w", err)
+		}
 		if err := o.handleHPA(hpaV2); err != nil {
 			klog.V(1).Infof("Encountered an error with the v2 HorizontalPodAutoscaler: %v. "+
 				"Falling back to try the v1 HorizontalPodAutoscaler", err)
+			// check if the HPA can be created using v1 API.
+			if ok, err := o.canCreateHPAV1(); !ok {
+				klog.V(1).Infof("The current configuration cannot be represented using v1 HorizontalPodAutoscaler.")
+				return fmt.Errorf("failed to create v2 HPA and the configuration is incompatible with v1: %w", err)
```

Comment:
No need for double the message, the error will bubble up to the user and that's sufficient information to present to the user.

Reply:
This still holds.